richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java

```diff
@@ -13,6 +13,28 @@
 
 import org.fxmisc.richtext.model.TwoDimensional.Position;
 
+/**
+ * This is one Paragraph of the document. Depending on whether the text is wrapped,
+ * it corresponds to a single line or it can also span multiple lines. A Paragraph
+ * contains of a list of SEG objects which make up the individual segments of the
+ * Paragraph. By providing a specific segment object and an associated segment
+ * operations object, all required data and the necessary operations on this data
+ * for a single segment can be provided. For example, {@code StyledText<S>} is a segment
+ * type which is used in the StyledTextArea, a text area which can render simple
+ * text only (which is already sufficient to implement all kinds of code editors).
+ *
+ * <p>For more complex requirements (for example, when images shall be part of the
+ * document) a different segment type must be provided (which can make use of
+ * {@code StyledText<S>} for the text part and add another segment type for images).
+ * <b>Note that Paragraph is an immutable class</b> - to modify a Paragraph, a new
+ * Paragraph object must be created. Paragraph itself contains some methods which
+ * take care of this, such as concat(), which appends some Paragraph to the current
+ * one and returns a new Paragraph.</p>
+ *
+ * @param <PS> The type of the paragraph style.
+ * @param <SEG> The type of the content segments in the paragraph (e.g. {@code StyledText<S>})
+ * @param <S> The type of the style of individual segments.
+ */
```

Comment:
This looks good. Perhaps there should also be a note that explains that constructing a `Paragraph` must always have at least one SEG object, even if it is an empty one, to insure the code works properly.